tests/compat.sh

```diff
@@ -216,14 +216,13 @@ filter_ciphersuites()
         G_CIPHERS=$( filter "$G_CIPHERS" )
     fi
 
-    # OpenSSL 1.0.1h doesn't support DTLS 1.2
-    if [ `minor_ver "$MODE"` -ge 3 ] && is_dtls "$MODE"; then
+    # OpenSSL <1.0.2 doesn't support DTLS 1.2. Check what OpenSSL
+    # supports from the s_server help. (The s_client help isn't
+    # accurate as of 1.0.2g: it supports DTLS 1.2 but doesn't list it.
+    # But the s_server help seems to be accurate.)
+    if ! $OPENSSL_CMD s_server -help 2>&1 | grep -q "^ *-$MODE "; then
+        M_CIPHERS=""
```

Comment:
If OpenSSL doesn't support a mode (in practice, DTLS 1.2), then we don't run `m->m`, `G->m` or `m->G` tests either. This is a bug. This commit allows runs with `$OPENSSL` (1.0.2) and `$OPENSSL_NEXT` (1.1.1) to test DTLS 1.2, as intended, but also accidentally removes m-m and G DTLS 1.2 tests when `$OPENSSL_LEGACY` is used. See https://github.com/ARMmbed/mbedtls/pull/5630/files#r826849976.